sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -100,6 +104,28 @@ private[parquet] class ParquetFilters(
     Binary.fromConstantByteArray(fixedLengthBytes, 0, numBytes)
   }
 
+  private def timestampValue(timestampType: TimestampLogicalTypeAnnotation, v: Any): JLong =
+    if (timestampType.getUnit == TimeUnit.MICROS) {
+      Option(v).map(
```

Comment:
Here within the `map` the if computes a transformation function. I think this is a bit complex and can be simplified especially as both branches of the outer if ends with an orNull `Option(v).map(...).orNull`.

I would suggest to first check whether v is null (you can keep using the `Option` with `map`) and just latter check for the TimeUnit.

A multiline `map` expression preferred to be written in the style:
```
map { v =>
  <transformation>
}
```

Reply:
Thanks, I also replaced the if to a match expression. With if, the code would silently convert NANO precision timestamps incorrectly, that's a bug, I think throwing an exception in this case is a better option.